lib/doorkeeper/models/access_token_mixin.rb

```diff
@@ -114,19 +118,24 @@ def find_access_token_in_batches(relation, **args, &block)
       #   Application instance
       # @param scopes [String, Doorkeeper::OAuth::Scopes]
       #   set of scopes
+      # @param custom_attributes [Nilable Hash]
+      #   A nil value, or hash with keys corresponding to the custom attributes
+      #   configured with the `custom_access_token_attributes` config option.
+      #   A nil value will ignore custom attributes.
       #
       # @return [Doorkeeper::AccessToken, nil] Access Token instance or
       #   nil if matching record was not found
       #
-      def find_matching_token(relation, application, scopes)
+      def find_matching_token(relation, application, custom_attributes, scopes)
```

Comment:
Do we need to pass `custom_attributes` here since we have it defined on a model level? Sorry it's hard to me to understand if we're passing some new attributes or the same as defined in the model (so can be find there already)

Reply:
I believe it's the custom attributes from the access grant retrieved [here](https://github.com/doorkeeper-gem/doorkeeper/blob/0aa94c5a82035ec4840785156760a2930e5de27a/lib/doorkeeper/oauth/authorization_code_request.rb#L105).

Reply:
Ah, OK, makes sense